smithy-model/src/main/java/software/amazon/smithy/model/node/DefaultNodeSerializers.java

```diff
@@ -0,0 +1,383 @@
+                    transientFields.add(StringUtils.uncapitalize(field.getName()));
+                }
+            }
+            return transientFields;
+        }
+
+        private static int getGetterPrefixCharCount(Method method) {
+            // Don't use static methods, or methods with arguments.
+            if (!Modifier.isStatic(method.getModifiers()) && method.getParameterCount() == 0) {
+                if (method.getName().startsWith("get")) {
+                    return 3;
+                } else if (method.getName().startsWith("is") && method.getReturnType() == boolean.class) {
+                    return 2;
+                }
+            }
+            return 0;
+        }
+    }
+
+    static final Serializer<Object> FROM_BEAN = new Serializer<Object>() {
+        @Override
+        public Class<Object> getType() {
+            return Object.class;
+        }
+
+        @Override
+        public Node serialize(Object value, Set<Object> serializedObjects, NodeMapper mapper) {
+            if (serializedObjects.contains(value)) {
+                return Node.nullNode();
+            }
+
+            // Add the current value to the set.
+            serializedObjects.add(value);
+            Map<StringNode, Node> mappings = new HashMap<>();
+            ClassInfo info = ClassInfo.fromClass(value.getClass());
+
+            for (Map.Entry<String, Method> entry : info.getters.entrySet()) {
+                try {
+                    Object getterResult = entry.getValue().invoke(value);
+                    Node result = mapper.serialize(getterResult, serializedObjects);
+                    if (mapper.getSerializeNullValues() || !result.isNullNode()) {
+                        mappings.put(Node.from(entry.getKey()), result);
+                    }
+                } catch (ReflectiveOperationException e) {
+                    // There's almost always a previous exception, so grab it's more useful message.
+                    // If this isn't done, I observed that the message of ReflectiveOperationException is null.
+                    String causeMessage = e.getCause() != null ? e.getCause().getMessage() : e.getMessage();
+                    String message = String.format(
+                            "Error serializing `%s` field of %s using %s(): %s",
+                            entry.getKey(),
+                            value.getClass().getName(),
+                            entry.getValue().getName(),
+                            causeMessage);
+                    throw new NodeSerializationException(message, e);
+                }
+            }
+
+            // Remove the current value from the set to ensure that it can be serialized
+            // multiple times (like in a List<T>).
+            serializedObjects.remove(value);
+
+            return new ObjectNode(mappings, SourceLocation.NONE);
+        }
+    };
+
+    // The priority-ordered list of default serializers.
```

Comment:
Minor: would be nice to know what determined this priority for any future updating.